crates/bevy_ecs/src/schedule_v3/executor/multi_threaded.rs

```diff
@@ -0,0 +1,573 @@
+
+            if !self.should_run(system_index, system, conditions, world) {
+                self.skip_system_and_signal_dependents(system_index);
+                continue;
+            }
+
+            self.running_systems.insert(system_index);
+            self.num_running_systems += 1;
+
+            if self.system_task_metadata[system_index].is_exclusive {
+                // SAFETY: `can_run` confirmed that no systems are running.
+                // Therefore, there is no existing reference to the world.
+                unsafe {
+                    let world = &mut *cell.get();
+                    self.spawn_exclusive_system_task(scope, system_index, systems, world);
+                }
+                break;
+            }
+
+            // SAFETY: No other reference to this system exists.
+            unsafe {
+                self.spawn_system_task(scope, system_index, systems, world);
+            }
+        }
+
+        // give back
+        self.ready_systems_copy = ready_systems;
+    }
+
+    fn can_run(
+        &mut self,
+        system_index: usize,
+        system: &mut BoxedSystem,
+        conditions: &mut Conditions,
+        world: &World,
+    ) -> bool {
+        #[cfg(feature = "trace")]
+        let _span = info_span!("check_access", name = &*system.name()).entered();
+
+        let system_meta = &self.system_task_metadata[system_index];
+        if system_meta.is_exclusive && self.num_running_systems > 0 {
+            return false;
+        }
+
+        if !system_meta.is_send && self.local_thread_running {
+            return false;
+        }
+
+        // TODO: an earlier out if world's archetypes did not change
+        for set_idx in conditions.sets_of_systems[system_index].difference(&self.evaluated_sets) {
+            for condition in &mut conditions.set_conditions[set_idx] {
+                condition.update_archetype_component_access(world);
+                if !condition
+                    .archetype_component_access()
+                    .is_compatible(&self.active_access)
+                {
+                    return false;
+                }
+            }
+        }
+
+        for condition in &mut conditions.system_conditions[system_index] {
+            condition.update_archetype_component_access(world);
+            if !condition
+                .archetype_component_access()
+                .is_compatible(&self.active_access)
+            {
+                return false;
+            }
+        }
+
+        if !self.skipped_systems.contains(system_index) {
+            system.update_archetype_component_access(world);
+            if !system
+                .archetype_component_access()
+                .is_compatible(&self.active_access)
+            {
+                return false;
+            }
+
+            // TODO: avoid allocation
+            self.system_task_metadata[system_index].archetype_component_access =
+                system.archetype_component_access().clone();
```

Comment:
Seems like we can avoid the allocation like this? Theres still slight room for optimization. FixedBitSet could in theory have a combined "clear and extend" operation that grows as necessary and then directly writes each piece of internal data from one set into the other. We could call it something like `a.set_to(&b)`.

@james7132: you're a FixedBitSet maintainer now, so this should be on your radar :)

```suggestion
            // PERF: use an optimized clear() + extend() operation
            let meta_access =
                &mut self.system_task_metadata[system_index].archetype_component_access;
            meta_access.clear();
            meta_access.extend(system.archetype_component_access());
```